apps/service_define/lib/define/event/store.ex

```diff
@@ -0,0 +1,85 @@
+defmodule Define.Event.Store do
+  require Logger
+  alias Define.DefinitionSerialization
+  alias Define.Model.{DataDefinitionView, ExtractView, LoadView, TransformView}
+  import Definition, only: [identifier: 1]
+
+  @instance Define.Application.instance()
+  @collection "definitions"
+
+  def update_definition(%Transform{} = data) do
+    update_definition_field(data, :transform, &to_transform_view/1)
+  end
+
+  def update_definition(%Load{} = data) do
+    update_definition_field(data, :load, &to_load_view/1)
+  end
+
+  def update_definition(%Extract{} = data) do
+    update_definition_field(data, :extract, &to_extract_view/1)
+  end
+
+  def update_definition(data) do
+    Logger.error(fn -> "Got unexpected data definition update: #{inspect(data)}" end)
+  end
```

Comment:
What's the spec of this function? Why is an unexpected input returning the same result as a good input?

Reply:
See my question about `:ok/:error` below. Same deal feels like it should apply here.

Reply:
I'm not quite sure I'm following what you're thinking here. By an unexpected input do you mean an unexpected event? `update_definition_field` ends up storing the event in the view state whereas the catch all does not. Are you wanting us to bring the storing in the viewstate up to these top level functions to make that more apparent?

Reply:
`Logger.error/1` returns `:ok` was what I was getting at.

If a valid definition is passed to `update_definition` but fails to persist, we log an error and return `:discard`.

If an invalid definition is passed to `update_definition`, we log an error and return `:ok`.